improve tutorial 4 recommend deploy contract first (#693)
o1-labs/docs2: docs/zkapps/tutorials/04-zkapp-ui-with-react.mdx

- The `export` statement creates a placeholder empty component.

```ts ignore
import { useEffect, useState } from 'react';
import './reactCOIServiceWorker';
import ZkappWorkerClient from './zkappWorkerClient';
import { PublicKey, Field } from 'o1js';
import GradientBG from '../components/GradientBG.js';
import styles from '../styles/Home.module.css';

let transactionFee = 0.1;
const ZKAPP_ADDRESS = 'B62qo2Be4Udo5EG1ux9yMJVkXe9Gz945cocN7Bn4W9DSYyeHZr1C3Ea';

export default function Home() {
  return <div/>
}
```

:::info

> For best results, deploy the zkApp to Berkeley yourself, as this test network is sometimes being reloaded. When deployed, replace `ZKAPP_ADDRESS` with the address of your deployment.

Reviewer: ```suggestion
The smart contract that the UI interacts with in this tutorial has been deployed to the Berkeley network and the public key is stored as the `ZKAPP_ADDRESS` in the UI project. If you experience problems with the deployed contract, you can deploy the `Add` contract included in the `contracts` folder yourself to the Berkeley network. When deployed, replace `ZKAPP_ADDRESS` with the  public key of your deployment.
```

> You will likely need to deploy the zkApp to Berkeley yourself, as this network is sometimes being reloaded. When deployed, replace the `ZKAPP_ADDRESS` with the address of your deployment.

Reviewer 2: ```suggestion
For best results, deploy the zkApp to Berkeley yourself, as this test network is sometimes being reloaded. When deployed, replace `ZKAPP_ADDRESS` with the address of your deployment.
```

:::

### Add state

This `export` statement creates mutable state that you can reference in the UI. The state updates as the application runs:

```ts ignore
...

> export default function Home() {

Reviewer 2: thanks for these updates @qwadratic 
I defer to  @ymekuria for the code update review and to verify that the tutorial runner does not have line number dependencies

Reviewer 2: @ymekuria  can you also evaluate https://github.com/o1-labs/docs2/issues/694
